supported-os-installer.sh

```diff
@@ -0,0 +1,524 @@
+	ADMIN_PASS="$(openssl rand -base64 48 | cut -c1-12)"
+	TIMEZONE="Asia/Singapore"
+	CURRENT_USER="$(whoami)"
+	sys_mem_kb="$(cat /proc/meminfo | grep MemAvailable | awk '{print $2}')"
+	SYSTEM_MEMORY_AVAIL="$(expr $sys_mem_kb / 1024 / 1024)"
+	# get version from tar file 10.0.0/8.0.0/9.0.0
+	ZCS_VERSION="$(echo $BUILD_ARTIFACT | grep -oP "(?<=-)\d+(\.\d+)+")" 
+	ver_pattern="10\.\([1-9]\|[1-9][0-9]\)\.*" # matches 10.1x.xx
+	INSTALL_PKGS=(
+		zimbra-ldap
+		zimbra-logger
+		zimbra-mta
+		zimbra-snmp
+		zimbra-store
+		zimbra-apache
+		zimbra-spell
+		zimbra-convertd
+		zimbra-memcached
+		zimbra-proxy
+		zimbra-archiving
+		zimbra-onlyoffice
+	)
+	if echo "$ZCS_VERSION" | grep $ver_pattern >> /dev/null; then
+		INSTALL_PKGS+=("zimbra-license-daemon")
+	fi
+
+	redhat_release="false"
+	command -v yum &>/dev/null
+	if [[ "$?" -eq 0 ]]; then
+		redhat_release="true";
+	fi
+
+	# Processing input
+	while [ $# -gt 0 ]; do
+		key="$1"
+		case $key in
+			-h|--help)
+			display_options
+			exit
+			;;
+			-d|--domain-name)
+			DOMAIN_NAME="$2"
+			shift
+			shift
+			;;
+			-i|--install-config)
+			INSTALL_CONFIG="$2"
+			shift
+			shift
+			;;
+			-b|--build-artifact)
+			BUILD_ARTIFACT="$2"
+			shift
+			shift
+			;;
+			-l|--license)
+			LICENSE="$2"
+			shift
+			shift
+			;;
+			-p|--admin-pass)
+			ADMIN_PASS="$2"
+			shift
+			shift
+			;;
+			-t|--time-zone)
+			TIMEZONE="$2"
+			shift
+			shift
+			;;
+			--optional-pkg)
+			INSTALL_PKGS+=("$2")
+			shift
+			shift
+			;;
+			*)
+			echo "Invalid argument(s): $2"
+			echo -e "\n\n\n"
+			display_options && exit 1
+			;;
+		esac
+	done
+
+	# Functions
+	check_vars
+	check_sudoer
+
+	# Install zimbra by injecting the config values.
+	install_zimbra
+}
+
+
+function check_vars {
+	[[ -z "$INSTALL_CONFIG" ]] && echo "Install defaults file path not defined." && exit 1;
+
+	# Check if license key of file is given, validate format
+	if [[ -f "$LICENSE" ]]; then
+		echo "License file is provided"
+		# Check if version is 10.1x.xx
+		if echo "$ZCS_VERSION" | grep $ver_pattern >> /dev/null; then
+			echo "$ZCS_VERSION Does not support license in file format. Use license key instead."
+			exit 1;
+		else 
+			license_file="$LICENSE"
+		fi
+	else
+		echo "License key is provided"
+		# Check if version is 10.1x.xx
+		if echo "$ZCS_VERSION" | grep $ver_pattern >> /dev/null; then
+			# Check if license key is in valid format
+			if echo "$LICENSE" | grep '^[0-9]\+$' >> /dev/null; then
+				license_key="$LICENSE"
+			else
+				echo "Invalid key format: $LICENSE License key must only contain numbers."
+				exit 1;
+			fi
+		else
+			echo "$ZCS_VERSION Does not support license key. Use license xml file format."
+			exit 1;
+		fi
+	fi
+
+	[[ ! -f "$BUILD_ARTIFACT" ]] && echo "Cannot locate the build artifact (zcs-*.tgz) file." && exit 1;
+	[[ -z "$HOSTNAME" ]] && echo "Cannot find \$HOSTNAME." && exit 1;
+	[[ -z "$ADMIN_PASS " ]] && echo "Admin password is empty. Using default password." && ADMIN_PASS="zimbra"
```

Comment:
should be random password

Reply:
True, Modified script to Generate random 12 characters password for admin if it is not provided as argument!